homeassistant/components/modbus/__init__.py

```diff
@@ -242,7 +242,13 @@ def setup(self):
         await self._connect_delay()
         async with self._lock:
             kwargs = {"unit": unit} if unit else {}
-            result = await func(address, count, **kwargs)
+            try:
+                result = await asyncio.wait_for(
```

Comment:
We can use async-timeout:
https://github.com/aio-libs/async-timeout/blob/master/README.rst

Reply:
Yes, but wait_for() not only does timeout but also cancels the transaction.

Are there any special disadvantages by using wait_for(), which does exactly what I want ?

Reply:
transaction in this case is the task.

Reply:
async-timeout will also cancel the task.

Its cleaner and faster than asyncio.wait_for and we use the former by convention.